pkg/build/strategies/sti/sti.go

```diff
@@ -228,16 +228,27 @@ func (builder *STI) Build(config *api.Config) (*api.Result, error) {
 	} else {
 		glog.V(1).Infof("Running %q in %q", api.Assemble, config.Tag)
 	}
-	if err := builder.scripts.Execute(api.Assemble, config.AssembleUser, config); err != nil {
+	startTime := time.Now()
+	err := builder.scripts.Execute(api.Assemble, config.AssembleUser, config)
+	builder.result.BuildInfo.Steps = utilstatus.RecordStepInfo(builder.result.BuildInfo.Steps, api.StepAssembleBuildScripts, startTime)
+	if err != nil {
 		if err == errMissingRequirements {
 			glog.V(1).Info("Image is missing basic requirements (sh or tar), layered build will be performed")
 			return builder.layered.Build(config)
 		}
-		if _, ok := err.(s2ierr.ContainerError); ok {
-			builder.result.BuildInfo.FailureReason = utilstatus.NewFailureReason(
-				utilstatus.ReasonAssembleFailed,
-				utilstatus.ReasonMessageAssembleFailed,
-			)
+		if e, ok := err.(s2ierr.ContainerError); ok {
+			if !isMissingRequirements(e.Output) {
+				builder.result.BuildInfo.FailureReason = utilstatus.NewFailureReason(
+					utilstatus.ReasonAssembleFailed,
+					utilstatus.ReasonMessageAssembleFailed,
+				)
+				return builder.result, err
+			}
+			glog.V(1).Info("Image is missing basic requirements (sh or tar), layered build will be performed")
```

Comment:
Question:  why not strive to instrument all steps so that "total duration of all the steps recorded should == the total duration of the build" ?